ggml-cuda.cu

```diff
@@ -2614,6 +2671,10 @@ void ggml_init_cublas() {
     static bool initialized = false;
 
     if (!initialized) {
+#ifdef GGML_USE_HIPBLAS
+    rocblas_initialize();
+    hipDeviceSynchronize();
+#endif
```

Comment:
As I said before, rocblas shouldn't be called directly at least not without gating it behind `__HIP_PLATFORM_AMD__`. Why is a rocblas import used just for this function alone? Either make the whole project use HIP entirely or ROC entirely, don't use both hipblas and rocblas. 

This should either be replaced by `hipblasInit` which is the equivalent hip call and the rocblas import removed if it should be called on all HIP supported platform. Or those rocblas stuff should be gated behind `__HIP_PLATFORM_AMD__` if they are only needed on AMD hardware. I'm guessing by laziness, the first option should be fine as I don't think a init and a sync call should impact performance too much.


Also, a minor nitpick but the indentation is not correct.

Finally, adding a comment for the upstream bug that make those change necessary should be added for later reader of that code. As without knowledge of the bug, this piece doesn't make sense. And this bug should be fixed in future ROCm versions, so it could/should be removed at some point.

Reply:
i was wrong. `hipblasInit` doesn't exist, it is referenced in [HIPIFY](https://github.com/ROCm-Developer-Tools/HIPIFY/blob/0b098927bf4f9788700191abd62ef26f782684f5/src/CUDA2HIP_BLAS_API_functions.cpp#L29), that why I suggesting it after doing a quick grep for `rocblas_initialize` in it, but in reality it doesn't exist, you can also see it in the line I had greped: `HIP_UNSUPPORTED`.

Reply:
The init should happen in `hipblasCreate` but I guess doing it before enumerating the devices can solve some issues when using multiple devices?